jest-puppeteer.config.js

```diff
@@ -1,6 +1,6 @@
 module.exports = {
   browserContext: 'incognito',
-  dumpio: true,
+  browserPerWorker: true,
```

Comment:
Found another useful speed up whilst debugging

We can change Puppeteer `setup()` → **`setup(jestConfig)`**

It'll then use a separate browser per test worker

Reply:
Sounds like this option will get us a long way. So from what I gathered (let me know if I misunderstood), Jest:
- spins a worker per test file, and with this new option, attach a pristine browser tab to it
- inside the worker runs tests from the file sequentially (prevents issues when tests rely on DB queries for ex), unless explicitely using `it.concurrent` (which [is experimental](https://jestjs.io/docs/api#testconcurrentname-fn-timeout) anyway and probably shouldn't use because of the shared tab)

It seems to me that we shouldn't have any tests stepping on each others toes with that. If it means passing the `page` to helper functions, that sounds pretty good to me.

Only question is whether it also starts one server per worker of if it's still a unique server running for all the tests (we don't really need more than one). I saw the code checking for live servers was updated a bit further up but don't know if it's for "general safety" or because we would now have multiple servers.

Reply:
Just one server

For any **JavaScript behaviour tests** project run (single, multiple, pattern, entire suite) It'll start once at the beginning of your `npx jest` run then close at the end 😊

This is because we've used "global setup" and "global teardown"

It's also true for `--watch` mode with the server starting up again when a file changes

I briefly looked into keeping the server running for the entire watch session until **Control + C** `SIGINT` [but there are bugs](https://github.com/smooth-code/jest-puppeteer/issues/305).

In terms of spinning up worker processes, I don't exactly know how it does it behind the scenes, but sounds about right 😆 

It'll vary sometimes as Jest reduces workers in `--watch` mode to keep CPU usage down, but now we're doing things the correct way we're all set up for any scenario, even `it.concurrent()` 🙌